frontend/packages/kubevirt-plugin/src/components/modals/scheduling-modals/affinity-modal/components/affinity-edit/affinity-edit.tsx

```diff
@@ -37,15 +37,21 @@ export const AffinityEdit: React.FC<AffinityEditProps> = ({
   onCancel,
 }) => {
   const [focusedAffinity, setFocusedAffinity] = React.useState(affinity);
+  const [affinityChanged, setAffinityChanged] = React.useState(false);
 
   const [
     affinityExpressions,
     ,
     onExpressionAdd,
-    onExpressionChange,
+    onLabelExpressionChange,
     onExpressionDelete,
   ] = useIDEntities<AffinityLabel>(affinity?.expressions);
 
+  const onExpressionChange = (updatedEntity: AffinityLabel) => {
```

Comment:
this behaviour should also be implemented on the node fields.

Reply:
+1

Reply:
\+ this could be implemented inside useIDEntities hook

Reply:
using useMemo :+1:

Reply:
now in useIDEntities :cake: